ecs-init/exec/iptables/iptables.go

```diff
@@ -213,10 +214,38 @@ func getBlockIntrospectionOffhostAccessInputChainArgs() []string {
 	}
 }
 
+// checkValidIfname checks that the string is a valid eth interface name. Returns
+// an error if the interface name is invalid.
+// see kernel dev_valid_name function for reference.
+// https://github.com/torvalds/linux/blob/d7e78951a8b8b53e4d52c689d927a6887e6cfadf/net/core/dev.c#L1056-L1079
+func checkValidIfname(name string) error {
+	if name == "" {
+		return fmt.Errorf("Invalid ifname [%s]: empty string not allowed", name)
+	}
+	if len(name) >= 16 {
+		return fmt.Errorf("Invalid ifname [%s]: length of name must be less than 16 chars", name)
+	}
+	if name == "." || name == ".." {
+		return fmt.Errorf("Invalid ifname [%s]: name or '.' or '..' not allowed", name)
```

Comment:
return fmt.Errorf("Invalid ifname [%s]: ~~name or~~ '.' or '..' not allowed", name)

Reply:
done